### ISSUE TITLE
Jetty 12 - Need hint that can set the HttpStream to close once the Request is complete

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#9179]
**Jetty version(s)**
Jetty 12

**Description**
While working on the `GracefulShutdownHandler`, it was discovered that there's no good way to close the `HttpStream` once the graceful response is complete.
Need a hint that can be accessed from the Request (somehow) to indicate to the Server that the active HttpStream should be closed once complete.

Using `Connection: close` is not a technique that can work on HTTP/2 or HTTP/3, and for Exchanges that are in committed state when this hint needs to be sent.

**Comment by gregw** [comment]
Duplicate of #9176 ?
